### ISSUE TITLE
Redshift & dbt: SQLFluff is not fixing errors (CV03, LT05)

### ISSUE THREAD

**Issue by jan-benisek** [sqlfluff/sqlfluff#5565]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I have a faulty dbt model which I want to fix. SQLFluff is triggered using a pre-commit.

### Expected Behaviour

I'd expect to fix these basic issues in the model - break the line and remove the comma.

### Observed Behaviour

This is the result after the `pre-commit` hook:
```shell
==== finding fixable violations ====
FORCE MODE: Attempting fixes...
== [smallpdf/models/gold/tools/calendar.sql] FAIL
L:  33 | P:   5 | LT05 | Line is too long (130 > 80).
                       | [layout.long_lines]
L:  33 | P: 131 | CV03 | Trailing comma in select statement forbidden
                       | [convention.select_trailing_comma]
  [1 templating/parsing errors found]
==== no fixable linting violations found ====
All Finished 📜 🎉!
  [2 unfixable linting violations found]
==== lint for unfixable violations ====
== [smallpdf/models/gold/tools/calendar.sql] FAIL
L:  33 | P: 131 | CV03 | Trailing comma in select statement forbidden
                       | [convention.select_trailing_comma]
L:  33 | P:   5 | LT05 | Line is too long (130 > 80). [layout.long_lines]
```

### How to reproduce

- dbt model:
```sql
{{
    config(
        schema='tools',
        materialized='table',
        dist='auto'
    )
}}


with

date_spine as (

    {{ dbt_utils.date_spine(
        datepart="day",
        start_date="to_date('01/01/2016', 'mm/dd/yyyy')",
        end_date="to_date('01/01/2029', 'mm/dd/yyyy')"
   )}}

)

select
    date_day,
    date_trunc('week', date_day) as date_week,
    date_trunc('month', date_day) as date_month,
    date_trunc('quarter', date_day) as date_quarter,
    date_trunc('year', date_day) as date_year,
    date_part('day', date_day) as day_num,
    date_part('week', date_day) as week_num,
    date_part('month', date_day) as month_num,
    date_part('year', date_day) as year_num,
    to_char(date_day, 'YYYY-MM-DD') as date_yyyymmdd,
    'this is a very long line' || 'becasue we want to test the line length' || 'and see if it is working' || 'or not' as long_line,

from date_spine
```
- then I run `git commit -am "foobar"` and I get the error mentioned above.

### Dialect

Redshift

### Version

sqlfluff, version 2.3.5
dbt, version 1.7.2
sqlfluff-templater-dbt, verison 2.3.5

### Configuration

- `pre-commit`
```
- repo: https://github.com/sqlfluff/sqlfluff
  rev: 2.3.5
  hooks:
    - id: sqlfluff-fix
      args: [--force, --ignore=templating]
      additional_dependencies: ['dbt-redshift==1.7.0', 'sqlfluff-templater-dbt==2.3.5']
```

- `.sqlfluff`
```
[sqlfluff]
verbose = 1
dialect = redshift
templater = jinja
output_line_length = 80
max_line_length = 80
sql_file_exts = .sql
large_file_skip_byte_limit = 50000

# RF04: Allow keywords to be used in SQL.
# ST06: Don't reorder columns.
# AL05: Unused aliases
# LT03: Trailing binary operators. Sometimes produces plain wrong results when
#       used in conjunction with Jinja
exclude_rules = RF04, ST06, LT02, AL05, LT03
warnings = none

[sqlfluff:templater:dbt]
project_dir = myproject/

[sqlfluff:indentation]
# See https://docs.sqlfluff.com/en/stable/layout.html#configuring-indent-locations
indent_unit = space
tab_space_size = 4
indented_joins = False
indented_ctes = False
indented_using_on = False
indented_on_contents = True
indented_then = True
indented_then_contents = False
allow_implicit_indents = False
template_blocks_indent = True

# If comments are found at the end of long lines, we default to moving them to
# the line _before_ their current location as the convention is that a comment
# precedes the line it describes.
trailing_comments = before

[sqlfluff:rules:capitalisation.keywords]
# Keywords.
capitalisation_policy = lower
# Comma separated list of words to ignore for this rule.
ignore_words = None
ignore_words_regex = None

[sqlfluff:rules:capitalisation.identifiers]
# Unquoted identifiers.
extended_capitalisation_policy = lower
# Comma separated list of words to ignore for this rule.
unquoted_identifiers_policy = aliases
quoted_identifiers_policy = none

[sqlfluff:rules:references.consistent]
# References must be consistently used.
force_enable = True

[sqlfluff:rules:capitalisation.functions]
extended_capitalisation_policy = lower

[sqlfluff:rules:capitalisation.literals]
capitalisation_policy = lower

[sqlfluff:rules:capitalisation.types]
# Data Types.
extended_capitalisation_policy = lower
# Comma separated list of words to ignore for this rule.
ignore_words = None
ignore_words_regex = None

[sqlfluff:templater]
unwrap_wrapped_queries = True

[sqlfluff:templater:jinja]
apply_dbt_builtins = True
load_macros_from_path =
    macros/foo,
    macros/bar,
    macros/tools

[sqlfluff:rules:convention.terminator]
# Semi-colon formatting approach.
multiline_newline = False
require_final_semicolon = False

[sqlfluff:rules:convention.casting_style]
preferred_type_casting_style = shorthand
casting_style = all

[sqlfluff:layout:type:comma]
#`touch`: Allows line breaks, but if no line break is present, then no space
# should be present.
spacing_before = touch
# Comma before (`leading`) or after (`trailing`) the statement.
line_position = trailing

[sqlfluff:layout:type:join_clause]
line_position = strict

[sqlfluff:rules:layout.functions]
# No space after function.
functions = all

[sqlfluff:rules:layout.long_lines]
ignore_comment_lines = False
ignore_comment_clauses = False

[sqlfluff:rules:layout.cte_bracket]
# Move the closing bracket on a new line.
cte_bracket = all
# Add a blank line after cte.
cte_newline = all

[sqlfluff:rules:ambiguous.column_references]
# `group by` and `order by` column references.
group_by_and_order_by_style = consistent

```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by peterbud** [comment]
Hi @jan-benisek , thanks for opening this issue.

The philosophy behind sqlfluff (and many similar linters) is generally to augment the author's intent rather than to take complete control of the layout. 

Minimal Intervention: The goal of a linter is to ensure code meets a specific standard (rules), not necessarily to rewrite it into a singular 'canonical' form. If your code satisfies the max_line_length and convention rules, the linter views both versions as "legal" expressions of that code.

Augmentation over Replacement: Think of the linter as a set of guardrails. It prevents you from drifting into "bad" territory (lines too long, inconsistent comma positioning), but it still leaves the "driving" (the specific aesthetic arrangement) to the developer.

Pls note that for CV03 the default is DISABLE trailing commas in select statement, see [CV03](https://docs.sqlfluff.com/en/stable/reference/rules.html#rule-CV03).

In your case if you want to _enable_ it, you can do with:

```ini
[sqlfluff:rules:convention.select_trailing_comma]
select_clause_trailing_comma=require
```
